src/seedu/addressbook/logic/Logic.java

```diff
@@ -85,7 +85,11 @@ public CommandResult execute(String userCommandText) throws Exception {
     private CommandResult execute(Command command) throws Exception {
         command.setData(addressBook, lastShownList);
         CommandResult result = command.execute();
-        storage.save(addressBook);
+        
+        if(command.isMutating()) {
```

Comment:
Does this follow Java Coding standards? Check [Layouts](https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html#layout).